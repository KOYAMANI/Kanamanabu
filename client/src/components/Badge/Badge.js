import React from 'react';
import { Card, Col } from 'react-bootstrap'



const Badge = ({achievement}) => (


  <Col xs={12} md={6} lg={4} key={achievement._id}>
    <Card style={{ width: '18rem' }}>
        <Card.Img variant="top" src={achievement.imageUrl} />
        <Card.Body>
            <Card.Title>{achievement.title}</Card.Title>
        </Card.Body>
    </Card>
  </Col>
)

export default Badge;