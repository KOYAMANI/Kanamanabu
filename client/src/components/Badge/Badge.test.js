import {render, screen, cleanup} from '@testing-library/react';
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Badge from "./Badge";

afterEach(cleanup);

const achievement =
    {
        _id: "1",
        title: 'achievement-title-test',
        imageUrl: 'https://google.com'
    }

describe('Achievement badge', () =>{

    test('Render without crashing', ()=>{
        const div = document.createElement('div')
        ReactDOM.render(<Badge achievement={achievement}/>, div)
    });

    test('Render title text', ()=>{
        render(<Badge achievement={achievement}/>)
        const element = screen.getByText('achievement-title-test')
        expect(element).toBeInTheDocument();
        expect(element).toHaveTextContent('achievement-title-test')
    });

    test('match snapshot',()=>{
        const tree = renderer.create(<Badge achievement={achievement}/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})