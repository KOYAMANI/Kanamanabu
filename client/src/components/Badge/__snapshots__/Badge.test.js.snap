// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Achievement badge match snapshot 1`] = `
<div
  className="col-lg-4 col-md-6 col-12"
>
  <div
    className="card"
    style={
      Object {
        "width": "18rem",
      }
    }
  >
    <img
      className="card-img-top"
      src="https://google.com"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        achievement-title-test
      </div>
    </div>
  </div>
</div>
`;
